src/memgraph.cpp

```diff
@@ -1200,20 +1204,35 @@ int main(int argc, char **argv) {
                             []() -> nlohmann::json { return query::plan::CallProcedure::GetAndResetCounters(); });
   }
 
+  communication::websocket::Server websocket_server{
+      boost::asio::ip::tcp::endpoint{boost::asio::ip::make_address("0.0.0.0"), 7444}};
+
+  {
+    auto sinks = spdlog::default_logger()->sinks();
+    sinks.push_back(websocket_server.GetLoggingSink());
+    spdlog::set_default_logger(std::make_shared<spdlog::logger>("memgraph_log", sinks.begin(), sinks.end()));
+  }
+
   // Handler for regular termination signals
-  auto shutdown = [&server, &interpreter_context] {
+  auto shutdown = [&websocket_server, &server, &interpreter_context] {
     // Server needs to be shutdown first and then the database. This prevents
     // a race condition when a transaction is accepted during server shutdown.
     server.Shutdown();
     // After the server is notified to stop accepting and processing
     // connections we tell the execution engine to stop processing all pending
     // queries.
     query::Shutdown(&interpreter_context);
+    websocket_server.Shutdown();
   };
-  InitSignalHandlers(shutdown);
 
   MG_ASSERT(server.Start(), "Couldn't start the Bolt server!");
+  websocket_server.Start();
+
+  InitSignalHandlers(shutdown);
```

Comment:
That is true, but IntepreterContext cannot receive queries without the Bolt Server running.
I could move it before starting WebSocket server but could be confusing.

Reply:
After some discussion on slack we figured out the following things:
1. The signal handler should be registered before the BOLT server is started, because otherwise (theoretically) the interpreter can be interrupted without gracefully shutting down (because of the default signal handler just terminates the application).
2. Furthermore in case of the application is terminated, the destructor of `Storage` won't be called, therefore the snapshots/wal file can end up in a corrupted state. 
3.  If the Memgraph receives SIGINT after the server was created and before it was started shutdown will be called on the server and then the server will be started. The start function creates a thread with infinite loop and the only way to stop it is from the shutdown, thus Memgraph won't shutdown in that case because the thread will never join.